plutus-contract/src/Language/Plutus/Contract/Effects/AwaitTxConfirmed.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE DerivingVia         #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ConstraintKinds     #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE DeriveAnyClass      #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE DerivingStrategies  #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE MonoLocalBinds      #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE OverloadedLabels    #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeOperators       #-}
+module Language.Plutus.Contract.Effects.AwaitTxConfirmed where
+
+import           Data.Aeson                                    (FromJSON, ToJSON)
+import           Data.Row
+import           Data.Set                                      (Set)
+import qualified Data.Set                                      as Set
+import           Data.Text.Prettyprint.Doc
+import           Data.Text.Prettyprint.Doc.Extras
+import           GHC.Generics                                  (Generic)
+import           Ledger                                        (TxId)
+
+import           IOTS                                          (IotsType)
+import           Language.Plutus.Contract.Request              (Contract, ContractRow, requestMaybe)
+import           Language.Plutus.Contract.Schema               (Event (..), Handlers (..), Input, Output)
+
+type TxConfirmationSym = "tx-confirmation"
```

Comment:
Worth taking a moment to check that this is the right terminology. What do other Cardano folk say for a transaction being finally added to the chain? Confirmed? Committed? Validated?

Also, we might want to distinguish between "validated" and "validated, and now won't be rolled back", but probably not at the moment.

Reply:
Nothing in the official style guide but I will ask some people

Reply:
I think "confirmed" is the appropriate term, even though it applies only when the transaction can't be rolled back anymore, not right away when it appears in the ledger.

Since we don't have rollbacks in the emulator I would be tempted to keep calling it "confirmed", and then make sure that the implementation in the real node client only produces the "confirmed" event when the tx can't be rolled back anymore.

Reply:
Cf. https://input-output-rnd.slack.com/archives/C2VJ41WDP/p1574869036041000

Reply:
Okay, seems fine.